### PR TITLE
Fix interactive updates

### LINKED ISSUE #3560: Plotter update broken?
### Describe the bug, what's wrong, and what you expected.

While trying to resolve #3380 I tried running the `chart_overlays` example which seems to be broken on the latest pyvista main branch. See below code sample to reproduce. I have been OOTL for a while, so I'm not sure if it's an actual bug in the interactor code or whether certain things have changed recently to produce these interactively updated plots.

### Steps to reproduce the bug.

I condensed the `chart_overlays` example to the following lines to reproduce the bug:

```py
import pyvista as pv


p = pv.Plotter()
p.add_mesh(pv.Sphere(1), name="sphere", render=False)
p.show(auto_close=False, interactive=True, interactive_update=True)  # Show plotter
# input("Press enter to elevate sphere.")
p.add_mesh(pv.Sphere(1, center=(0, 0, 10)), name="sphere", render=False)
p.update()

p.show()  # Keep plotter open at end
```

Yielding following trace on my machine:

```
Traceback (most recent call last):
  File "C:/Users/bramd/AppData/Roaming/JetBrains/PyCharm2021.2/scratches/scratch_4.py", line 9, in <module>
    p.update()
  File "P:\python\pyvista\pyvista\plotting\plotting.py", line 1943, in update
    self.right_timer_id = self.iren.create_repeating_timer(stime)
  File "P:\python\pyvista\pyvista\plotting\render_window_interactor.py", line 780, in create_repeating_timer
    self.interactor.DestroyTimer(timer_id)
AttributeError: 'NoneType' object has no attribute 'DestroyTimer'

Process finished with exit code 1
```

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Sun Nov 06 17:38:28 2022 West-Europa (standaardtijd)

                OS : Windows
            CPU(s) : 8
           Machine : AMD64
      Architecture : 64bit
               RAM : 15.9 GiB
       Environment : Python
       File system : NTFS
        GPU Vendor : Intel
      GPU Renderer : Intel(R) HD Graphics 530
       GPU Version : 4.4.0 - Build 21.20.16.4550

  Python 3.7.0 (v3.7.0:1bf9cc5093, Jun 27 2018, 04:59:51) [MSC v.1914 64 bit
  (AMD64)]

           pyvista : 0.38.dev0
               vtk : 9.2.0
             numpy : 1.21.5
           imageio : 2.13.1
            scooby : 0.5.9
             pooch : v1.6.0
        matplotlib : 3.4.3
             PyQt5 : 5.15.6
           IPython : 7.30.1
          colorcet : 3.0.0
           cmocean : 2.0
        ipyvtklink : 0.2.1
             scipy : 1.7.3
        itkwidgets : 0.32.1
              tqdm : 4.62.3
            meshio : 4.4.6
        jupyterlab : 3.2.4
         pythreejs : Version unknown
--------------------------------------------------------------------------------
```

### Screenshots

_No response_

### LINKED ISSUE #3584: Creating movies not working
### Describe the bug, what's wrong, and what you expected.

After I started using python 3.10 with pyvista 0.37 my scripts to create mp4 movies stopped working. Not even the example in the documentation is running ([Create a MP4 Movie](https://docs.pyvista.org/examples/02-plot/movie.html) ) due to the same error. This is the stacktrace I got from both my script and the example:

```
Traceback (most recent call last):
  File "/home/rafael/...", line 39, in <module>
    plotter.write_frame()  # Write this frame
  File "/home/rafael/.pyenv/versions/.../lib/python3.10/site-packages/pyvista/plotting/plotting.py", line 4365, in write_frame
    self.update()
  File "/home/rafael/.pyenv/versions/.../lib/python3.10/site-packages/pyvista/plotting/plotting.py", line 1943, in update
    self.right_timer_id = self.iren.create_repeating_timer(stime)
  File "/home/rafael/.pyenv/versions/.../lib/python3.10/site-packages/pyvista/plotting/render_window_interactor.py", line 780, in create_repeating_timer
    self.interactor.DestroyTimer(timer_id)
AttributeError: 'NoneType' object has no attribute 'DestroyTimer'
```

Does anyone know how to solve or have any workaround for it? Go back to a previous version of python is not an option for me. Thank you all in advance!

### Steps to reproduce the bug.

just try to execute the example

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Fri Nov 11 00:49:07 2022 -03

                OS : Linux
            CPU(s) : 16
           Machine : x86_64
      Architecture : 64bit
       Environment : Python
        GPU Vendor : NVIDIA Corporation
      GPU Renderer : NVIDIA GeForce GTX 1660 Ti/PCIe/SSE2
       GPU Version : 4.5.0 NVIDIA 520.56.06

  Python 3.10.8 (main, Nov  7 2022, 20:05:38) [GCC 12.2.0]

           pyvista : 0.37.0
               vtk : 9.2.2
             numpy : 1.23.4
           imageio : 2.22.4
            scooby : 0.7.0
             pooch : v1.6.0
        matplotlib : 3.6.2
         pyvistaqt : 0.9.0
             PyQt5 : 5.15.7
             scipy : 1.9.3
--------------------------------------------------------------------------------
```

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by dcbr**
### Overview

Fix interactive updates, resolves #3560, resolves #3584.
Also fixes an error when closing the plotter window while chart interaction was enabled.

### Details

- #3245 introduced the line `self.interactor = None` in the `terminate_app` method. The `on_timer` callback only needs a `TerminateApp()` call (to resume execution from a blocking `Start` call), without actually destroying the interactor itself. This PR moves the `self.interactor = None` line to the `close` method instead. I hope this is fine, but it would be nice if the contributors of #3245 could chime in :)
- I also tried to improve the documentation a bit for the related "interactive update" code.
- Another change in #3245 is to call the `remove_observers` method from the active `_style_class` in the `close` method. While this works for all interaction classes created from the `_style_factory`, it does not work for the `vtkContextInteractorStyle` class used for chart interaction. This caused an error while closing the plotter window when this chart interaction is enabled. My quick fix here was to first disable chart interaction before further cleaning up.

### CODE REVIEW

**Comment by dcbr** [comment]
So, the timer test fails on Linux for VTK<9.2 because the ProcessEvents method did not properly process timer events, this was fixed in [!8560](https://gitlab.kitware.com/vtk/vtk/-/merge_requests/8560) and released in VTK9.2. I have no access to a Mac OS system, so not sure what is causing the fails there. If it is using another interactor than `vtkXRenderWindowInteractor` it might have the same bug that is still unfixed in VTK9.2.
Other than that this PR is ready for review now.

**Comment by dcbr** [comment]
@pyvista/developers this is ready for review.

The codecov complaints are not relevant IMO:

- Lines 1950-1954 and 5904 in `plotting.py` have always been there and are now only executed for older VTK versions.
- The mentioned lines in `render_window_interactor.py` are all tested in `test_timer` but skipped on some (older) VTK versions (maybe that's why there is a complaint?).

File: tests/test_render_window_interactor.py [hunk 1]
```diff
@@ -99,3 +102,53 @@ def callback(mouse_point):
...
+    platform.system() == 'Darwin',
+    reason='vtkCocoaRenderWindowInteractor (MacOS) does not invoke TimerEvents during ProcessEvents. ',
+)
+@pytest.mark.needs_vtk_version(
+    (9, 2),
+    reason='vtkXRenderWindowInteractor (Linux) does not invoke TimerEvents during ProcessEvents until VTK9.2.',
+)
+def test_timer():
+    # Create normal interactor from offscreen plotter (not generic,
+    # which is the default for offscreen rendering)
```
Comment (tkoyama010) [inline tests/test_render_window_interactor.py:117]:
```suggestion
    # Create a normal interactor from the offscreen plotter (not generic,
    # which is the default for offscreen rendering)
```

File: tests/test_render_window_interactor.py [hunk 2]
```diff
@@ -99,3 +102,53 @@ def callback(mouse_point):
...
+        # Helper function to call process_events for the given duration (in milliseconds).
+        t = 1000 * time.time()
+        while 1000 * time.time() - t < duration:
+            iren.process_events()
+
+    # Setup interactor
+    iren.add_observer("TimerEvent", on_timer)
+    iren.initialize()
+
+    # Test one-shot timer (only fired once for extended duration)
```
Comment (tkoyama010) [inline tests/test_render_window_interactor.py:140]:
```suggestion
    # Test one-shot timer (only fired once for the extended duration)
```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

**Review by tkoyama010 — APPROVED** [review]

**Comment by dcbr** [comment]
@pyvista/developers could we move along with this?

**Comment by banesullivan** [comment]
Hey, yes, I will merge this today. I assigned myself as review to double check this issues you were seeing on Mac.

This is fairly critical so it needs to merge

**Review by banesullivan — APPROVED** [review]
Things are working on my M2 and Intel Macs. I was able to reproduce the error on `main` and no longer able to on this branch.

I'm merging `main` into this to run CI one more time and ensure passing then will merge.

**Pull request merged**